lib/hanami/slice.rb

```diff
@@ -219,6 +219,7 @@ def prepare_slice
 
         prepare_all
 
+        instance_exec(container, &@prepare_container_block) if @prepare_container_block
```

Comment:
This is basically the change, moving this instance_exec to a different spot. Everything else in this PR is ensuring we now have appropriate test coverage for this behaviour :)